scripts/source-maintenance.sh

```diff
@@ -15,39 +15,52 @@
 ## Source Code Format Enforcement
 #
 # A checker to recursively reformat all source files: .h .c .cc .cci
-# using a custom astyle formatter and to use MD5 to validate that
-# the formatter has not altered the code syntax.
+# using a custom astyle formatter and to use md5sum (or similar) to validate
+# that the formatter has not altered the code syntax.
 #
 # If code alteration takes place the process is halted for manual intervention.
 #
 
-# whether to continue execution after a failure
 # TODO: Expand the subset of failures covered by this feature; see run_().
 KeepGoing="no"
 # the actual name of the directive that enabled keep-going mode
 KeepGoingDirective=""
-#
-# The script checks that the version of astyle is TargetAstyleVersion.
-# if it isn't, the default behaviour is to not perform the formatting stage
-# in order to avoid unexpected massive changes if the behaviour of astyle
-# has changed in different releases.
-# if --with-astyle /path/to/astyle is used, the check is still performed
-# and a warning is printed, but the sources are reformatted
+
+# the version of astyle tool required by Squid coding style
 TargetAstyleVersion="3.1"
-ASTYLE='astyle'
 
 # whether to check and, if necessary, update boilerplate copyright years
 CheckAndUpdateCopyright=yes
 
 printUsage () {
-    echo "Usage: $0 [option...]"
-    echo "options:"
-    echo "    --keep-going|-k"
-    echo "    --check-and-update-copyright <yes|no>"
-    echo "    --with-astyle </path/to/astyle/executable>"
+cat <<_EOF
+Usage: $0 [option...]
+options:
+    --keep-going|-k                            (default: stop on error)
+    --check-and-update-copyright <yes|no>      (default: no)
+    --with-astyle </path/to/astyle/executable> (default: astyle)
+    --only-changed-since <fork|commit-id>      (default: apply to all files)
+
+This script applies Squid mandatory code style guidelines.
+
+It requires astyle version $TargetAstyleVersion, or it will skip formatting
+program files. The path to the astyle binary can be specified using the
+--with-astyle option or with the ASTYLE environment variable.
+
+It will try to auto-detect a checksum program (e.g. md5sum).
+
+If the --only-changed-since argument is supplied, it expects a git commit-id,
+branch name or the special keyword 'fork'. The script will try identifying
+changed files since the specified commit and, if successful, only examine
+files that have changed. The keyword 'fork' will look for files changed
+since the current branch was forked off 'upstream/master'. Sensible values
+for this argument may include HEAD^, master, origin/master, or the branch
+the current one was forked off
+_EOF
 }
```

Comment:
If I remember correctly I considered changed-only mostly as an optimization, not as a hard promise.
Should I make this explicit?

Reply:
I am not sure a "yes" or "no" answer to your question would help you address my change request because I do not think the `--only-changed-since` option should be described or implemented as "mostly an optimization" (regardless of your original intentions or considerations).

The option name implies and the option primary documentation promises that only changed things will be affected or considered. An _ideal_ implementation of that option should result in no changes to unmodified files (or even unmodified lines!). Until we reach that ideal, we should disclose implementation limitations; that disclosure is what my change request is all about. We may never be able to fully reach that ideal, of course, but reaching it should be the goal, and goals matter.

Please see the original change request for specific suggestions, but I do not insist on any specific wording.

Reply:
> I am not sure a "yes" or "no" answer to your question would help you address my change request because I do not think the `--only-changed-since` option should be described or implemented as "mostly an optimization" (regardless of your original intentions or considerations).
> 
> The option name implies and the option primary documentation promises that only changed things will be affected or considered. An _ideal_ implementation of that option should result in no changes to unmodified files (or even unmodified lines!). Until we reach that ideal, we should disclose implementation limitations; that disclosure is what my change request is all about. We may never be able to fully reach that ideal, of course, but reaching it should be the goal, and goals matter.
> 
> Please see the original change request for specific suggestions, but I do not insist on any specific wording.
> 
> While you are adjusting the text, please end this documentation paragraph with a dot/period.

Done